src/bt_peer_connection.cpp

```diff
@@ -1991,7 +1991,7 @@ namespace libtorrent
 		// only, since they might disconnect immediately when
 		// they have downloaded a single piece, although we'll
 		// make another piece available
-		detail::write_uint8(t->is_upload_only() && !t->super_seeding(), ptr);
+		detail::write_uint8(std::uint8_t(t->is_upload_only() && !t->super_seeding()), ptr);
```

Comment:
maybe I misunderstood the warning. msvc was saying that a uint8 was being implicitly cast to a bool (which I believe would happen in ``write_uint8()`` when passed a ``bool``)